asimov

```diff
@@ -76,33 +58,37 @@ exclude_file() {
     done
 }
 
-# Start to construct the `find` parameter arguments, based on an initial directory.
-declare -a find_parameters=("${ASIMOV_ROOT}")
-
 # Iterate over the skip directories to construct the `find` expression.
+declare -a find_parameters_skip=()
 for d in "${ASIMOV_SKIP_PATHS[@]}"; do
-    find_parameters+=( -not \( -path "${d}" -prune \) )
+    find_parameters_skip+=( -not \( -path "${d}" -prune \) )
 done
 
-# Iterate over dependencies.
-declare n=0
+# Iterate over the directory/sentinel pairs to construct the `find` expression.
+declare -a find_parameters_vendor=()
 for i in "${ASIMOV_VENDOR_DIR_SENTINALS[@]}"; do
     read -ra parts <<< "${i}"
 
     # Add this folder to the `find` list, allowing a single `find` command to find all
-    if [ "$n" -gt "0" ]; then
-        find_parameters+=(-o)
-    fi
-    find_parameters+=(-name "${parts[0]}" -type d -prune)
```

Comment:
When I submitted this PR I'd already forgotten what the original problem was, but I think this was it. The original code was pruning (not descending any farther) when a directory name matched, but *before* checking if the sentinel file existed.